src/net/connect.hpp

```diff
@@ -68,10 +70,15 @@ void connect_base(std::string address, int port, Callback<bufferevent *> cb,
         return;
     }
 
-    // WARNING: set callbacks after connect() otherwise we free `bev` twice
-    // NOTE: In case of `new` failure we let the stack unwind
```

Comment:
Yes you are right, somehow I thought that in refactoring to do the free inside of the callback we would have ended up avoiding the double free, but you are right that this issue still would arise.